internal/testutil/testutil.go

```diff
@@ -168,8 +194,18 @@ func kptfileExists(t *testing.T, path string) bool {
 // this set is not guaranteed to contain all differing files.
 func Diff(sourceDir, destDir string) (sets.String, error) {
 	// get set of filenames in the package source
+	sourceDir, clean1, err := addMergeCommentToExpected(sourceDir)
```

Comment:
True. I made it configurable and added addMergeCommentToSource as option to the Diff method. Not adding comments to dest as it is expected to have comments. So this would test that the comments are added for cases where comments should be added.

Callers like get_test, update_test would pass true for addMergeCommentToSource as comments should be added after these operations
Callers like fetch_test, resource-merge_test don't are elemental and don't add merge comments. So addMergeCommentToSource is false for them when repos are compared.